server/api/routes.ts

```diff
@@ -220,47 +259,47 @@ function pack(promise, res, next) {
     });
 }
 
-function getUser(req) {
+async function getUser(req) {
   // Get username
-  return LDAP.getActiveUser(req);
+  let temp = await LDAP.getActiveUser(req);
+  return temp;
 }
 
 /*
-  Note: No longer using in OSS version but being left for future features
   levels of access:
     admin: administrators
     approver: can approve new projects
     anon: unknown users with minimal access
 */
-type AccessTypes =
-  'admin'
-  | 'approver'
-  | 'anon';
-
-export async function checkAccess(user): Promise<AccessTypes> {
+enum AccessTypes {
+  admin = 'admin',
+  approve = 'approver',
+  anon = 'anon',
+}
+// Middleware that stuffs user access levels into req.UserAccess
+export async function checkAccess(req, res, next) {
+  let user = await getUser(req);
   // Checks that the user has access to their requested information
   let groups = await LDAP.getGroups(user);
-  let access;
+  let access = [];
   let len = config.admin.posixGroup.length;
+  // Check if in admin groups
   for (let i = 0; i < len; i++) {
     if ((groups as any).includes(config.admin.posixGroup[i])) {
-      access = 'admin';
+      access.push('admin');
     }
   }
-  if (!access) {
-    if ((groups as any).includes(config.approver.posixGroup)) {
-      access = 'approver';
-    } else {
-      access = 'anon';
+  len = config.approver.posixGroup.length;
+  // Check if in approver groups
+  for (let i = 0; i < len; i++) {
+    if ((groups as any).includes(config.approver.posixGroup[i])) {
+      access.push('approver');
     }
   }
-  return access;
-}
-
-export function approvedAccess(type, list): boolean {
-  if (list.includes(type)) {
-    return true;
-  } else {
-    return false;
+  // If access is empty then they aren't a member of a special group
+  if (access.length === 0) {
+    access.push('anon');
```

Comment:
Any benefit to this versus leaving the array empty?

Reply:
I wanted to make sure the user is always categorized so any additions we/people add later can depend on the consistency.